### PR TITLE
feat(forwardProps): add forwardProps option

### PR DESCRIPTION
**Pull request by kentcdodds**
**What**: This adds a `forwardProps` option to the `glamorous` function

**Why**: So if you're wrapping a component in `glamorous`, you can specify some props to forward, regardless of the `rootEl` you define.

**How**: TDD! Also I had to update some dependencies to get rid of annoying warnings from React :)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -111,13 +111,21 @@ function glamorous(comp, {rootEl, displayName} = {}) {
 
     Object.assign(
       GlamorousComponent,
-      getGlamorousComponentMetadata({comp, styles, rootEl, displayName}),
+      getGlamorousComponentMetadata({
+        comp,
+        styles,
+        rootEl,
+        forwardProps,
```
Comment (kentcdodds) [inline src/index.js]: This is the added property, other than that it's formatting.

File: src/index.js [hunk 2]
```diff
@@ -111,13 +111,21 @@ function glamorous(comp, {rootEl, displayName} = {}) {
...
+        displayName,
+      }),
     )
     return GlamorousComponent
   }
 }
 
-function getGlamorousComponentMetadata({comp, styles, rootEl, displayName}) {
+function getGlamorousComponentMetadata(
+  {comp, styles, rootEl, forwardProps, displayName},
```
Comment (kentcdodds) [inline src/index.js]: just added `forwardProps` here

File: src/index.js [hunk 3]
```diff
@@ -227,7 +236,10 @@ function splitProps(
   }
   return Object.keys(rest).reduce(
     (split, propName) => {
-      if (shouldForwardProperty(rootEl, propName)) {
+      if (
+        forwardProps.indexOf(propName) !== -1 ||
```
Comment (kentcdodds) [inline src/index.js]: Here's the added check. Remarkably simple :)

**Comment by kentcdodds** [comment]
Any collaborator can feel free to squash and merge this PR after it's been reviewed :+1: You shouldn't need to change the commit message.

**Comment by paulmolluzzo** [comment]
I'm on the beach without a keyboard until Sunday. 🏝😎🍹

**Comment by kentcdodds** [comment]
Nice!

**Comment by kentcdodds** [comment]
I just realized that we don't have any of the `options` (`rootEl`, `displayName`, and now `forwardProps`) documented :scream: I'd really like to get this merged first. If someone else wants to document those that'd be awesome. Or I can do it later.

Gotta run! 🏃

**Comment by kentcdodds** [comment]
I added documentation in the `README` for all the props because I had time :)

**Comment by kentcdodds** [comment]
I need this in my app, so I'm going to merge this. Feel free to add reviews/comments to it and we can fix it in post :+1: Thanks friends!

**Comment by kkemple** [comment]
I meant to do an official review, i went through your comments and it all looked good to me! :shipit:

**Pull request merged**